### DISCUSSION TITLE
Compiling the Firmware? (originally, Manual Leveling)

### DISCUSSION THREAD

**Discussion by usma11pb** [aegean-odyssey/mpmd_marlin_1.1.x#71] [Q&A] [answered]
I have made some significant modifications to my mpmd, and for reasons that would be too long to go into here, I need to switch to manual bed leveling.  I am not the greatest with coding though, and I am at a loss as to how to compile the binary file to upload to the mpmd.  I am assuming I don't have the board settings correct.  I have tried VScode and Arduino, but exporting from Arduino creates two .hex files and I haven't figured out what VScode is producing.  I've followed the steps from the marlinfw website, but I get stuck at the last step.  Can anyone in this setting confirm that the board setting we need in the config.h is the Ramps_14_EFB for the stock mpmd? If so, has anyone here had luck with exporting to a .bin for upload?  

I have attempted to follow the steps provided in the Q&A regarding the make file, but again I'm very new to coding and I was lost in the sauce.  Like lots of sauce.  

Any help would be greatly appreciated.

**Comment by aegean-odyssey** [comment, marked as answer]
___IF___ you are using the printer's stock controller board and the `mpmd_marlin_1.1.x` firmware here, then there are a few ways compile the code. There is some discussion in #66 and #55.

I think possibly the easiest "development environment" to set up would be to install a "git" program and a favorite text editor on your PC, and then:
* clone the aegean/mpmd_marlin_1.1.x repository to your github account;
* use git to clone your repository to a folder on your PC;
* use your favorite editor to make changes;
* use git to push your changes to your github repository;
* (via a browser) use the github "Actions" to compile the firmware; and
* (via a browser) download the artifact (zip file) -- it contains the "bin" file.

>  If your changes are not too elaborate, then the editing can be done via a web browser working directly on the clone of the repository in your github account, so nothing needs to be installed on your PC. This will likely be too cumbersome for any substantial changes, so using git and an editor on your local PC is probably a better choice.


You can update firmware in the printer by placing the firmware on the micro SD card. More info is in the [wiki](/aegean-odyssey/mpmd_marlin_1.1.x/wiki/Installing-the-Firmware).

As far as a board setting for `config.h`, the `mpmd_marlin_1.1.x` firmware defines a new board type (and files) specifically for the controller board found in the stock Monoprice MP Mini Delta printer.

**Reply by usma11pb** [reply]
This worked.  Better than any possible hope and dreams I could have hoped for.  Thank you so much!
